src/main/java/pl/edu/pw/elka/paprykaisalami/geeruh/issues/adapters/persistent/IssuePersistent.java

```diff
@@ -114,18 +146,33 @@ public Issue toIssue() {
                 .summary(new Summary(summary))
                 .description(new Description(description))
                 .assigneeUserId(assigneeUserId)
+                .relatedIssues(relatedIssues
+                        .stream()
+                        .map(ri -> new IssueId(
+                                new ProjectCode(ri.getProject().getCode()), ri.getIssueIndex())
```

Comment:
I would make a function which returns a domain `IssueId` from `IssuePersistent` and use it here (especially that it needs to be used twice)